src/app/test_executive/gossip_consistency.ml

```diff
@@ -0,0 +1,120 @@
+open Core
+open Integration_test_lib
+open Signature_lib
+
+module Make (Inputs : Intf.Test.Inputs_intf) = struct
+  open Inputs
+  open Engine
+  open Dsl
+
+  (* TODO: find a way to avoid this type alias (first class module signatures restrictions make this tricky) *)
+  type network = Network.t
+
+  type node = Network.Node.t
+
+  type dsl = Dsl.t
+
+  let block_producer_balance = "1000" (* 1_000_000_000_000 *)
+
+  let config =
+    let n = 5 in
+    let open Test_config in
+    { default with
+      requires_graphql= true
+    ; block_producers=
+        List.init n ~f:(fun _ ->
+            {Block_producer.balance= block_producer_balance; timing= Untimed}
+        ) }
+
+  let expected_error_event_reprs = []
+
+  let pk_of_keypair keypairs n =
+    Public_key.compress (List.nth_exn keypairs n).Keypair.public_key
+
+  let wait_for_all_to_initialize ~logger network t =
+    let open Malleable_error.Let_syntax in
+    let producers = Network.block_producers network in
+    let n = List.length producers in
+    List.mapi producers ~f:(fun i node ->
+        let%map () = wait_for t (Wait_condition.node_to_initialize node) in
+        [%log info] "Block producer %d (of %d) initialized" (i + 1) n ;
+        () )
+    |> Malleable_error.all_unit
+
+  let send_payments ~logger ~sender_pub_key ~receiver_pub_key ~amount ~fee
+      ~node n =
+    let open Malleable_error.Let_syntax in
+    let rec go n =
+      if n = 0 then return ()
+      else
+        let%bind () =
+          Network.Node.send_payment ~retry_on_graphql_error:false ~logger
+            ~sender_pub_key ~receiver_pub_key ~amount ~fee node
+        in
+        go (n - 1)
+    in
+    go n
+
+  let wait_for_payments ~dsl ~sender_pub_key ~receiver_pub_key ~amount n =
```

Comment:
@nholland94 so I'm trying to do as you suggested in the other comment and wait for the payemts to get included in the chain instead of a hardcoded timeout, is this the right approach?  will there be weirdness when we're waiting for 10 different transactions all with the same sender receiver and amount?